Dockerfile

```diff
@@ -11,19 +11,20 @@
 #  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 #  License for the specific language governing permissions and limitations under
 #  the License.
+
 ARG DOCKER_ORG="usdotfhwastoldev"
 ARG DOCKER_TAG="develop"
 FROM ${DOCKER_ORG}/carma-base:${DOCKER_TAG} as base_image
 FROM base_image as setup
 ARG GIT_BRANCH="develop" 
+ENV LD_LIBRARY_PATH="/opt/ros/foxy/lib:${LD_LIBRARY_PATH}"
```

Comment:
This is a great catch.
Summarizing the issue - the sourced environment without this change seems to include noetic and foxy installations in turn causing issues with launching and running tests for cpp_message.
The need to do this kind of hints that there is something wrong with how the packages are being built/launched.
 
Looking at this a little more, it seems like this is being caused the entrypoint script in carma-base which by default sources noetic if it exists. https://github.com/usdot-fhwa-stol/carma-base/blob/c5aa3edad836e082d8d4905a08650f1a7623d532/noetic-foxy/init-env.sh#L43

The fix should probably go there instead.